packages/xchain-binance/src/client.ts

```diff
@@ -198,37 +193,29 @@ class Client implements BinanceClient, XChainClient {
    * @throws {"Invalid phrase"}
    * Thrown if the given phase is invalid.
    */
-  setPhrase = (phrase: string): Address => {
-    if (!this.phrase || this.phrase !== phrase) {
-      if (!xchainCrypto.validatePhrase(phrase)) {
-        throw new Error('Invalid phrase')
-      }
-
-      this.phrase = phrase
-      this.privateKey = null
-      this.address = ''
+  setPhrase = (phrase: string, index = 0): Address => {
+    if (!validatePhrase(phrase)) {
+      throw new Error('Invalid phrase')
     }
 
-    return this.getAddress()
+    this.phrase = phrase
+    return this.getAddress(index)
   }
 
   /**
    * @private
    * Get private key.
    *
+   * @param {number} index account index for the derivation path
    * @returns {PrivKey} The privkey generated from the given phrase
    *
    * @throws {"Phrase not set"}
    * Throws an error if phrase has not been set before
    * */
-  private getPrivateKey = (): PrivKey => {
-    if (!this.privateKey) {
-      if (!this.phrase) throw new Error('Phrase not set')
-
-      this.privateKey = crypto.getPrivateKeyFromMnemonic(this.phrase)
-    }
+  private getPrivateKey = (index: number): PrivKey => {
+    if (!this.phrase) throw new Error('Phrase not set')
```

Comment:
^ Why another check ^? Existing `phrase` has been already checked in constructor and `setPhrase` and  should always be there

Reply:
done